tests/virt_autotest/virt_utils.pm

```diff
@@ -714,8 +714,10 @@ sub cleanup_host_and_guest_logs {
     #Clean dhcpd and named services up explicity
     if (get_var('VIRT_AUTOTEST')) {
         script_run("brctl addbr br123;brctl setfd br123 0;ip addr add 192.168.123.1/24 dev br123;ip link set br123 up");
-        script_run("service dhcpd restart") if (script_run("systemctl restart dhcpd") ne '0');
-        script_run("service named restart") if (script_run("systemctl restart named") ne '0');
+        if (!get_var('VIRT_UEFI_GUEST_INSTALL')) {
+            script_run("service dhcpd restart") if (script_run("systemctl restart dhcpd") ne '0');
+            script_run("service named restart") if (script_run("systemctl restart named") ne '0');
+        }
```

Comment:
Is it proper to have service related operations in function cleanup_host_and_guest_logs which aims for logs?

Reply:
Currently, I can think of two ways: 1) change the function name to indicate better what's included in it 2) put the service related into a new function, then call it in this function

Reply:
Actually there is no need to change function name which already reflects what is  included. I already explained in the last comment. Option 2) is better compared with 1) although it needs some code change. And what option 2) can benefit most is a standalone subroutine to help perform operations on services. Actually I intended to do so but had no decision. So if you like, I can make the change.

Reply:
[Verification](http://10.67.129.106/tests/1293#step/uefi_guest_verification/13) also done.